controllers/windowsmachine/windowsmachine_controller.go

```diff
@@ -218,6 +222,33 @@ func isWindowsMachine(labels map[string]string) bool {
 	return false
 }
 
+func isValidMachine(obj runtime.Object) bool {
+	machine := &mapi.Machine{}
+
+	// If for some reason this function called on an object which is not a Node, return
+	if kind := obj.GetObjectKind().GroupVersionKind(); kind.Kind != machine.Kind {
+		return false
+	}
+
+	var ok bool
+	machine, ok = obj.(*mapi.Machine)
+	log.Info(fmt.Sprintf("%+v\n", machine))
+	if !ok {
+		log.Info("unable to typecast object to machine:\n%s", fmt.Sprintf("%+v\n", obj))
+		return false
+	}
+	if machine.Status.Phase == nil {
+		log.Info("machine object has no phase associated with it:\n%s", fmt.Sprintf("%+v\n", machine.Name))
```

Comment:
so use `log.Debug()`?

Reply:
I don't `log.Debug()` exists. We have been using in `log.V(1).Info()` instead.

Reply:
additionally the log message should be of the format log.V(1).Info("machine object has no phase associated with it" , "name", machine.Name) else it will throw odd number of key value pairs error.